common/scala/src/main/scala/org/apache/openwhisk/core/database/elasticsearch/ElasticSearchActivationStore.scala

```diff
@@ -0,0 +1,482 @@
+
+class ElasticSearchActivationStore(
+  httpFlow: Option[Flow[(HttpRequest, Promise[HttpResponse]), (Try[HttpResponse], Promise[HttpResponse]), Any]] = None,
+  elasticSearchConfig: ElasticSearchActivationStoreConfig =
+    loadConfigOrThrow[ElasticSearchActivationStoreConfig](ConfigKeys.elasticSearchActivationStore),
+  useBatching: Boolean = false)(implicit actorSystem: ActorSystem,
+                                actorMaterializer: ActorMaterializer,
+                                logging: Logging)
+    extends ActivationStore
+    with MultipleReadersSingleWriterCache[WhiskActivation, DocInfo] {
+
+  import com.sksamuel.elastic4s.http.ElasticDsl._
+
+  private implicit val executionContextExecutor: ExecutionContextExecutor = actorSystem.dispatcher
+
+  private val httpClientCallback = new HttpClientConfigCallback {
+    override def customizeHttpClient(httpClientBuilder: HttpAsyncClientBuilder): HttpAsyncClientBuilder = {
+      val provider = new BasicCredentialsProvider
+      provider.setCredentials(
+        AuthScope.ANY,
+        new UsernamePasswordCredentials(elasticSearchConfig.username, elasticSearchConfig.password))
+      httpClientBuilder.setDefaultCredentialsProvider(provider)
+    }
+  }
+
+  private val client =
+    ElasticClient(
+      ElasticProperties(s"${elasticSearchConfig.protocol}://${elasticSearchConfig.hosts}"),
+      NoOpRequestConfigCallback,
+      httpClientCallback)
+
+  private val esType = "_doc"
+  private val maxOpenDbRequests = actorSystem.settings.config
+    .getInt("akka.http.host-connection-pool.max-connections") / 2
+  private val batcher: Batcher[IndexRequest, Either[ArtifactStoreException, DocInfo]] =
+    new Batcher(500, maxOpenDbRequests)(doStore(_)(TransactionId.dbBatcher))
+
+  private val minStart = 0L
+  private val maxStart = Instant.now.toEpochMilli + 100L * 365 * 24 * 60 * 60 * 1000 //100 years from now
+
+  override def store(activation: WhiskActivation, context: UserContext)(
+    implicit transid: TransactionId,
+    notifier: Option[CacheChangeNotification]): Future[DocInfo] = {
+
+    val key = CacheKey(activation)
+    cacheUpdate(activation, key, doStore(activation))
+  }
+
+  private def doStore(activation: WhiskActivation)(implicit transid: TransactionId): Future[DocInfo] = {
+    val start =
+      transid.started(this, LoggingMarkers.DATABASE_SAVE, s"[PUT] 'activations' document: '${activation.docid}'")
+
+    val path = activation.annotations
+      .getAs[String](WhiskActivation.pathAnnotation)
+      .getOrElse(s"${activation.namespace}/${activation.name}")
+    // Escape `_id` field as it's not permitted in ElasticSearch, add `path` field for search, and
+    // convert annotations to JsObject as ElasticSearch doesn't support array with mixed types
+    val response = JsObject(
+      "statusCode" -> JsNumber(activation.response.statusCode),
+      "result" -> JsString(activation.response.result.toJson.compactPrint))
+    val payload = JsObject(
```

Comment:
Just to confirm ... the response is present 2 times

1. As stored field
2. As indexed document

Reply:
it's only used as a stored field